Userland/Libraries/LibWeb/Infra/Numbers.cpp

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (c) 2023, Jonatan Klemets <jonatan.r.klemets@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/GenericLexer.h>
+#include <LibWeb/Infra/CharacterTypes.h>
+#include <LibWeb/Infra/Numbers.h>
+
+namespace Web::Infra {
```

Comment:
Since the algorithms here come from the HTML spec, they should go in the `Web::HTML` namespace (and directory).